realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -160,15 +159,25 @@ public void onChange(AllJavaTypes object) {
         parent.setFieldObject(child);
         looperThreadRealm.commitTransaction();
 
-        assertEquals(2, looperThreadRealm.where(AllJavaTypes.class).findAll().size());
-        assertEquals(1, counter.get());
+        // This should be on the Looper queue, behind the notification
+        looperThread.postRunnable(
```

Comment:
Got it. So the call to `beginTransaction` triggers notification of what, exactly.  The documentation (https://realm.io/docs/java/latest/#realm-notifications) seems to indicate that I can get notifications for a specific object.  I think you may be saying that every listener in the realm gets notified, when I start a transaction on the realm.